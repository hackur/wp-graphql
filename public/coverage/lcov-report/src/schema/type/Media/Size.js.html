<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/schema/type/Media/Size.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">src/schema/type/Media</a> Size.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/3</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17</td><td class="line-coverage quiet"><span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { Grap<span class="cstat-no" title="statement not covered" >hQLObjectType, GraphQLString, GraphQLInt } from 'graphql';<span class="cstat-no" title="statement not covered" ></span></span>
&nbsp;
const MediaSizeType <span class="cstat-no" title="statement not covered" >= new GraphQLObjectType({</span>
  name: 'MediaSize',
  description: 'The details for the media size.',
  fields: {
    name: { type: GraphQLString },
    width: { type: GraphQLInt },
    height: { type: GraphQLInt },
    file: { type: GraphQLString },
    mime_type: { type: GraphQLString },
    source_url: { type: GraphQLString },<span class="cstat-no" title="statement not covered" ></span>
  },
});
&nbsp;
export default MediaSizeType;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Jun 30 2017 19:47:20 GMT-0400 (EDT)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
