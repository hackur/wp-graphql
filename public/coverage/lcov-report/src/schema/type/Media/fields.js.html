<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/schema/type/Media/fields.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">src/schema/type/Media</a> fields.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/18</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/11</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40</td><td class="line-coverage quiet"><span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { Grap<span class="cstat-no" title="statement not covered" >hQLNonNull, GraphQLID } from 'graphql';<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ></span></span></span></span>
import { toGlobalId <span class="cstat-no" title="statement not covered" >} from 'graphql-relay';</span>
import { dat<span class="cstat-no" title="statement not covered" >e, modified } from 'field/da</span>te';
import metaF<span class="cstat-no" title="statement not covered" >ield from 'field/meta';<span class="cstat-no" title="statement not covered" ></span></span>
import { globalIdF<span class="cstat-no" title="statement not covered" >ield, slug, guid, link } from 'fie</span>ld/identifier';
import { title <span class="cstat-no" title="statement not covered" >} from 'field/content';</span>
import { typ<span class="cstat-no" title="statement not covered" >e, template } from 'field/po</span>st';
import { comme<span class="cstat-no" title="statement not covered" >ntStatus, pingStatus } from 'f</span>ield/status';
import { desc<span class="cstat-no" title="statement not covered" >ription, caption, altText, me</span>diaType, mimeType, sourceUrl } from 'field/media';
import author <span class="cstat-no" title="statement not covered" >from 'field/author';<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ></span></span></span></span></span>
&nbsp;
export default {
  id: globalIdField(),
  ...date,
  ...guid,
  ...modified,
  ...slug,
  ...type,
  ...link,
  ...title,
  ...commentStatus,
  ...pingStatus,
  ...template,
  ...author,
  meta: metaField(),
  // featured_media (this field should exist for audio/video)
  // extra media fields
  ...description,
  ...caption,
  ...altText,
  ...mediaType,
  ...mimeType,
  ...sourceUrl,
  post: {
    type: new GraphQLNonNull(GraphQLID),
    description: 'The ID for the associated post of the attachment.',
    resolve: data =&gt; t<span class="fstat-no" title="function not covered" >oGlobalI</span>d('Post<span class="cstat-no" title="statement not covered" >', data.post),</span>
  },
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Jun 30 2017 19:47:20 GMT-0400 (EDT)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
