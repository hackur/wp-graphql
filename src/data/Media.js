import { toGlobalId } from 'graphql-relay';
import Dataloader from 'dataloader';
import fetchData from 'data/utils';

// Dataloader expects IDs that can be read by the REST API

const path = process.env.WP_MEDIA_ENDPOINT || 'wp/v2/media';
const mediaLoader = new Dataloader(ids =>
  fetchData(path, {
    qs: { include: ids, orderby: 'include', per_page: 100 },
  }).then(({ data: { body } }) => body)
);
const slugLoader = new Dataloader(slugs =>
  fetchData(path, { qs: { slug: slugs, orderby: 'slug', per_page: 100 } }).then(
    ({ data: { body } }) => body
  )
);

class Media {
  getID() {
    return toGlobalId(this.constructor.name, this.id);
  }

  static getEndpoint() {
    return path;
  }

  static async load(id) {
    const data = await mediaLoader.load(id);
    return data ? Object.assign(new Media(), data) : null;
  }

  static async loadBySlug(slug) {
    const data = await slugLoader.load(slug);
    return data ? Object.assign(new Media(), data) : null;
  }
}

export default Media;
