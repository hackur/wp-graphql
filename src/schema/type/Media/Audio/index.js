import { GraphQLObjectType } from 'graphql';

import MediaInterface from 'interface/Media';
import mediaFields from 'type/Media/fields';
import AudioDetails from 'type/Media/Audio/Details';

const AudioType = new GraphQLObjectType({
  name: 'Audio',
  description: 'An object.',
  interfaces: [MediaInterface],
  isTypeOf(media) {
    return media.mime_type.indexOf('audio') === 0;
  },
  fields: {
    ...mediaFields,
    mediaDetails: { type: AudioDetails },
  },
});

export default AudioType;
