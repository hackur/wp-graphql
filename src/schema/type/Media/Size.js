import { GraphQLObjectType, GraphQLString, GraphQLInt } from 'graphql';

const MediaSizeType = new GraphQLObjectType({
  name: 'MediaSize',
  description: 'The details for the media size.',
  fields: {
    name: { type: GraphQLString },
    width: { type: GraphQLInt },
    height: { type: GraphQLInt },
    file: { type: GraphQLString },
    mimeType: { type: GraphQLString },
    sourceUrl: { type: GraphQLString },
  },
});

export default MediaSizeType;
