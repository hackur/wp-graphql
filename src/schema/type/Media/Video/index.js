import { GraphQLObjectType } from 'graphql';

import MediaInterface from 'interface/Media';
import mediaFields from 'type/Media/fields';
import VideoDetails from 'type/Media/Video/Details';

const VideoType = new GraphQLObjectType({
  name: 'Video',
  description: 'An object.',
  interfaces: [MediaInterface],
  isTypeOf(media) {
    return media.mime_type.indexOf('video') === 0;
  },
  fields: {
    ...mediaFields,
    mediaDetails: { type: VideoDetails },
    resolve: media => media.media_details,
  },
});

export default VideoType;
