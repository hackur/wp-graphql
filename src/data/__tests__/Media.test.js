import Media from 'data/Media';

describe('Test Media data access', () => {
  test('Get endpoint', () => {
    expect(Media.getEndpoint()).toMatchSnapshot();
  });

  test('Load a media item', async () => {
    const media = await Media.load(13);
    expect(media.getID()).toMatchSnapshot();
    expect(media).toMatchSnapshot();
  });
});
