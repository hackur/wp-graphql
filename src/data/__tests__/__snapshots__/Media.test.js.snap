// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Media data access Get endpoint 1`] = `"wp/v2/media"`;

exports[`Test Media data access Load a media item 1`] = `"TWVkaWE6MTM="`;

exports[`Test Media data access Load a media item 2`] = `
Media {
  "id": 13,
}
`;
